classes/class-p4-taxonomy-page.php

```diff
@@ -0,0 +1,146 @@
+		 */
+		public function add_taxonomy_form_fields( $wp_tag ) {
+
+			$args = [
+				'hierarchical'     => 1,
+				'post_type'        => 'page',
+				'show_option_none' => 'None',
+				'name'             => 'category_page',
+				'id'               => 'category-page-id',
+				'class'            => 'category-page-class',
+			];
+
+			// Filter issue pages
+			$options          = get_option( 'planet4_options' );
+			$explore_page_id  = $options['explore_page'];
+			if ( $explore_page_id ) {
+			    $args['child_of'] = $explore_page_id;
+			}
+
+			if ( isset( $wp_tag ) && $wp_tag instanceof WP_Term ) {
+				$category_page = get_term_meta( $wp_tag->term_id, 'category_page', true );
+				if ( $category_page ) {
+					$args['selected'] = $category_page;
+				} ?>
+				<tr class="form-field edit-wrap term-category-page-wrap">
+					<th>
+						<label><?php echo __( 'Select Category Page', 'planet4-master-theme' ); ?></label>
+					</th>
+					<td>
+						<?php wp_dropdown_pages( $args ); ?>
+						<p class="description"><?php echo __( 'Associate this category with a page.', 'planet4-master-theme' ); ?></p>
+					</td>
+				</tr>
+			<?php } else { ?>
+				<div class="form-field add-wrap term-category-page-wrap">
+					<label><?php esc_html_e( 'Select Category Page', 'planet4-master-theme' ); ?></label>
+					<?php wp_dropdown_pages( $args ); ?>
+					<p class="description"><?php esc_html_e( 'Associate this category with a page.', 'planet4-master-theme' ); ?></p>
+				</div>
+				<?php
+			}
+		}
+
+		/**
+		 * Save taxonomy custom field(s).
+		 *
+		 * @param int $term_id The ID of the WP_Term object that is added or edited.
+		 */
+		public function save_taxonomy_meta( $term_id ) {
+			$field_id = 'category_page';
+			$category_page = filter_input( INPUT_POST, $field_id, FILTER_VALIDATE_INT );
+
+			if ( $this->validate( $category_page ) ) {
+				update_term_meta( $term_id, $field_id, $category_page );
+			}
+		}
+
+		/**
+		 * Add custom column.
+		 *
+		 * @param array $columns Associative array with the columns of the taxonomy.
+		 *
+		 * @return array Associative array with the columns of the taxonomy.
+		 */
+		public function edit_taxonomy_columns( $columns ) : array {
+			$columns['category_page'] = __( 'Category Page', 'planet4-master-theme' );
+			return $columns;
+		}
+
+		/**
+		 * Apply custom output to a custom column.
+		 *
+		 * @param string $output The html to be applied to each row of the $column.
+		 * @param string $column The name of the column to be managed.
+		 * @param int    $term_id The ID of the WP_Term object that is added or edited.
+		 *
+		 * @return string The new html to be applied to each row of the $column.
+		 */
+		public function manage_taxonomy_custom_column( $output, $column, $term_id ) : string {
+			if ( 'category_page' === $column ) {
+				$category_page = get_term_meta( $term_id, 'category_page', true );
+				$output        = get_the_title( $category_page );
```

Comment:
May I suggest we put a link to the Edit page of the relevant page here instead of just the title, in order to make better use of this column.
We could output:
`				$output        = '<a href="' . get_edit_post_link( $category_page ) . '" target="_blank">' . get_the_title( $category_page ) . '</a>';
`

Reply:
Agreed